### ISSUE TITLE
Clean-up feature for Scipion projects

### ISSUE THREAD

**Issue by JuhaHuiskonen** [I2PC/scipion#2078]
The file number in large projects can become hard to manage.  It would make sense to have a feature to delete those files that are not registered as outputs required by Scipion and that are not generally needed.

Each protocol could define files:
- that are necessary for Scipion workflow (will never be deleted)
- files that are useful in some cases outside of Scipion but not used in the workflow (deleted only if 'full' clean is selected, e.g. those diagnostic files that can't be visualised by Scipion)
- intermediate files that are very rarely needed (deleted if 'quick' or 'full' clean is selected, e.g. iterations 2...N-1)

The user could then select all the protocols they want to clean and select either of the two options (quick or full).

**Comment by pconesa** [comment]
Thanks Juga. We clearly need to do something. What I'm not sure is if it's
more hassle for the developers to flag all possible output files or just
delete those that are not final, like all iteration files in iterative
protocols.

**Comment by JuhaHuiskonen** [comment]
The protocols that increase the file count the most seem to be RELION's particle polishing and CTF refinement (especially in projects with many many movies - as these protocols create several diagnostic files for each movie). Refinement and classification runs produces significantly fewer files.

**Comment by pconesa** [comment]
@azazellochg , @delarosatrevin , It might be a good time now maybe to review the amount of files relion keeps. If you have the chance, rethink if files kept are really useful for a regular user.

I'm mention here @dmaluenda to be aware of this for xmipp methods.

**Comment by delarosatrevin** [comment]
Yes, I plan to have a look at it...but maybe it would be good time to
define one cleanUp method in base Protocol class...and make then actionable
from the GUI. So it is ready for when each protocol decides to implement
it. I have some ideas for a general implementation. We could discuss about
it.

**Comment by pconesa** [comment]
Anything that will make life easy for developers is welcome.
